syft/pkg/cataloger/generic/lookup.go

```diff
@@ -0,0 +1,85 @@
+package generic
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"regexp"
+	"text/template"
+
+	"github.com/anchore/syft/internal"
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/file"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger/internal/unionreader"
+	"github.com/anchore/syft/syft/source"
+)
+
+// Lookup is a generic package lookup that can be used to match a package definition
+// To a file that meets the given content criteria of the EvidencePatternTemplates.
+type Lookup struct {
+	Package string
+	// FilepathPatterns is a list of regular expressions that will be used to match against the file path of a given
+	// source location. If any of the patterns match, the file will be considered a candidate for parsing.
+	FilepathPatterns []*regexp.Regexp
+	// EvidencePatternTemplates is a list of regular expressions that will be used to match against the file contents of a
+	// given file in the source location. If any of the patterns match, the file will be considered a candidate for parsing.
+	EvidencePatternTemplates []string
```

Comment:
Just noting the comment from below that maybe this should be a callback function?

Reply:
Added a longer explanation to the main comment